// Copyright (c) 2014, Cloudera, inc.
//
// Protobuf used for introspection of RPC services (eg listing in-flight RPCs,
// reflection, etc)

package kudu.rpc;

import "rpc/rpc_header.proto";

message RpcCallInProgressPB {
  required RequestHeader header = 1;
  optional string trace_buffer = 2;
  optional uint64 micros_elapsed = 3;
}

message RpcConnectionPB {
  required string remote_ip = 1;
  // TODO: swap out for separate fields
  required string remote_user_credentials = 2;
  repeated RpcCallInProgressPB calls_in_flight = 3;
}

message DumpRunningRpcsRequestPB {
  optional bool include_traces = 1 [ default = false ];
}

message DumpRunningRpcsResponsePB {
  repeated RpcConnectionPB inbound_connections = 1;
  repeated RpcConnectionPB outbound_connections = 2;
}